riposte-core/src/main/java/com/nike/riposte/server/handler/RequestInfoSetterHandler.java

```diff
@@ -47,13 +59,31 @@ else if (msg instanceof HttpContent) {
                                                     + "HttpProcessingState. This should be impossible");
                 }
 
-                requestInfo.addContentChunk((HttpContent) msg);
+                int currentRequestLengthInBytes = requestInfo.addContentChunk((HttpContent) msg);
+                int configuredMaxRequestSize = getConfiguredMaxRequestSize(state);
+                
+                if (!isMaxRequestSizeValidationDisabled(configuredMaxRequestSize) && currentRequestLengthInBytes > configuredMaxRequestSize) {
+                    throw new TooLongFrameException("Request raw content length exceeded configured max request size of " + configuredMaxRequestSize);
+                }
             }
         }
 
         return PipelineContinuationBehavior.CONTINUE;
     }
 
+    private boolean isMaxRequestSizeValidationDisabled(int configuredMaxRequestSize) {
+        return configuredMaxRequestSize <= 0;
+    }
+
+    private int getConfiguredMaxRequestSize(HttpProcessingState state) {
+        int maxConfiguredRequestSize = maxRequestSizeInBytes;
+        Optional<Integer> endpointMaxRequestSizeOverride = state.getEndpointForExecution().maxRequestSizeInBytesOverride();
```

Comment:
After some more thinking about all this, here are some general refactors that I think should be done:

1. The max request size in bytes (MRSB) was originally an int because once upon a time we used Netty's `HttpObjectAggregator`, and *that* handler did the MRSB check and required an int. We no longer have that restriction, so we should use a long instead to get rid of the 2GB ceiling on MRSB.
2. Since we have to check for null on the endpoint override anyway even with `Optional`, and the core stuff tends to use nulls rather than optionals to avoid the perf hit of optionals, the `Endpoint.maxRequestSizeInBytesOverride()` method signature should probably change to just return a `Long` object. Returning null from the override method would mean "use the global default", and returning anything non-null would mean "use this value instead of the global default".